diff --git a/configure b/configure
index 28f3eb0..5dc6947 100755
--- a/configure
+++ b/configure
@@ -478,9 +478,9 @@ echo "CXX=$cxx" >> $config_host_mak
 print_config "CXX" "$cxx"

 # generate io_uring_version.h
-MAKE_PRINT_VARS="include Makefile.common\nprint-%: ; @echo \$(\$*)\n"
-VERSION_MAJOR=$(env echo -e "$MAKE_PRINT_VARS" | make -s --no-print-directory -f - print-VERSION_MAJOR)
-VERSION_MINOR=$(env echo -e "$MAKE_PRINT_VARS" | make -s --no-print-directory -f - print-VERSION_MINOR)
+VERSION=$(awk '/Version:/ { print $2 }' liburing.spec)
+VERSION_MAJOR=$(echo ${VERSION} | cut -d. -f1)
+VERSION_MINOR=$(echo ${VERSION} | cut -d. -f2)
 io_uring_version_h="src/include/liburing/io_uring_version.h"
 cat > $io_uring_version_h << EOF
 /* SPDX-License-Identifier: MIT */