diff --git a/cmake/patch-imported-config.cmake b/cmake/patch-imported-config.cmake
index ec68d74..96eef0d 100644
--- a/cmake/patch-imported-config.cmake
+++ b/cmake/patch-imported-config.cmake
@@ -20,11 +20,6 @@ if(TARGET c-ares::cares)
   project_build_tools_patch_default_imported_config(c-ares::cares)
 endif()
 
-# curl targets
-if(TARGET CURL::libcurl)
-  project_build_tools_patch_default_imported_config(CURL::libcurl)
-endif()
-
 # abseil targets
 if(WITH_ABSEIL)
   project_build_tools_patch_default_imported_config(
